### PR TITLE
[SHOP-1365] Video modal unable to close

### PR DESCRIPTION
**Pull request by tlkiong**
[Click here for story](https://sephora-asia.atlassian.net/browse/SHOP-1365)

### CODE REVIEW

**Comment by kieudongoctruc** [comment]
what is the root cause of it? `a` is no longer supported? I see some other PRs also seem to tackle this issue. have you take a look?
https://github.com/sephora-sea/sephora-style-guide/pull/143
https://github.com/sephora-sea/sephora-style-guide/pull/130

**Comment by tlkiong** [comment]
@kieudongoctruc I didn't see the other PRs yet
Yeah. It seems like `a` is not supported.

---

Checking the other 2 PRs, I prefer my way here because it isn't dependent on external to determine if its the right player ID or not.

File: app/assets/javascripts/sephora_style_guide/video.js [hunk 1]
```diff
@@ -4,37 +4,37 @@ var players = [];
...
 // this function gets called when API is ready to use
 function onYouTubeIframeAPIReady() {
-  var allPlayers = $('.modal-video').children('iframe');
-
-  for (var el in Array.from(allPlayers)) {
-    var player = new YT.Player($(allPlayers[el]).attr('id'), {
+  $('.modal-video').children('iframe').each(function() {
+    const id = $(this).attr('id')
+  
+    let player = new YT.Player(id, {
```
Comment (kieudongoctruc) [inline app/assets/javascripts/sephora_style_guide/video.js:10]: const?

File: app/assets/javascripts/sephora_style_guide/video.js [hunk 2]
```diff
@@ -4,37 +4,37 @@ var players = [];
...
+    const currentVideoId = $(this).attr('data-video');
 
-    for (var i in players) {
-      if (players[i].a.id === currentVideoId) {
-        players[i].playVideo();
-      }
-    }
+    players.find(player => {
+      return player.id === currentVideoId
+    })?.player?.playVideo()
```
Comment (kieudongoctruc) [inline app/assets/javascripts/sephora_style_guide/video.js:30]:
```suggestion
    players.find(player => player.id === currentVideoId)?.player?.playVideo()
```

File: app/assets/javascripts/sephora_style_guide/video.js [hunk 3]
```diff
@@ -4,37 +4,37 @@ var players = [];
...
+    const currentVideoId = $(this).find('iframe').attr('id');
 
-    for (var i in players) {
-      if (players[i].a.id === currentVideoId) {
-        players[i].pauseVideo();
-      }
-    }
+    players.find(player => {
+      return player.id === currentVideoId
+    })?.player?.pauseVideo()
```
Comment (kieudongoctruc) [inline app/assets/javascripts/sephora_style_guide/video.js:38]:
```suggestion
    players.find(player => player.id === currentVideoId)?.player?.pauseVideo()
```

File: app/assets/javascripts/sephora_style_guide/video.js (hunk 1 above)
Comment (tlkiong) [reply app/assets/javascripts/sephora_style_guide/video.js:10]: Good catch

**Review by kieudongoctruc — APPROVED** [review]

**Pull request merged**